pydantic/types.py

```diff
@@ -175,19 +175,34 @@ def validate(cls, value, values, **kwarg):
         return make_dsn(**kwargs)
 
 
-class ConstrainedInt(int):
+class ConstrainedMeta(type):
```

Comment:
Maybe better to change name of class? Something like `ConstrainedNumberMeta` or just `NumberMeta`?